src/plugins/janus_sip.c

```diff
@@ -1771,6 +1772,127 @@ static void janus_sip_sofia_logger(void *stream, char const *fmt, va_list ap) {
 	}
 }
 
+/* Sofia SIP logger function: when the Event Handlers mechanism is enabled,
+ * we use this to intercept SIP messages sent by the stack (received
+ * messages are more easily recoverable in janus_sip_sofia_callback) */
+static void janus_sip_sofia_logger_single(void *stream, char const *fmt, va_list ap) {
+	if(!fmt)
+		return;
+	char line[255];
+	char message[16000];
+	char sofia_log_buf[16000];
+	char call_id_hdr[255];
+	sofia_log_buf[0] = '\0';
+	call_id_hdr[0] = '\0';
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#ifndef __clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=format"
+#endif
+	g_vsnprintf(line, sizeof(message), fmt, ap);
+#ifndef __clang__
+#pragma GCC diagnostic pop
+#endif
+#pragma GCC diagnostic pop
+	gchar **parts = g_strsplit(message, "\n", -1);
+	if (parts) {
+		int index = 0;
+		char *line = NULL, *cr = NULL;
+		while((line = parts[index]) != NULL) {
+			cr = strchr(line, '\r');
+			if(cr != NULL)
+				*cr = '\0';
+			if(*line == '\0') {
+				if(cr != NULL)
+					*cr = '\r';
+				index++;
+				continue;
+			}
+
+			/* First line of the log indicates if we send or receive a message. */
+			/* We are only interested in the sent ones */
+			if(index==0) {
+				if(strstr(line, "send ") == line) {
+					int length = atoi(line+5);
+					JANUS_LOG(LOG_HUGE, "Intercepting message (%d bytes)\n", length);
+					index++;
+					continue;
+				}
+				g_strfreev(parts);
+				return ;
+			}
+
+			if(strlen(line) == 1) {
+				/* Append a carriage and return */
+				janus_strlcat(sofia_log_buf, "\r\n", sizeof(sofia_log_buf));
+			} else if(strstr(line, "-") == line ) {
+				index++;
+				continue;
+			} else if(strstr(line, "OPTIONS") == line) {
+				index++;
+				continue;
+			} else {
+				/* Is this a Call-ID header? Keep note of it */
+				if(strstr(line, "Call-ID") == line) {
+					g_snprintf(call_id_hdr, sizeof(call_id_hdr), "%s", line+9);
+				}
+				/* Append the line to our buffer, skipping the indent */
+				janus_strlcat(sofia_log_buf, line, sizeof(sofia_log_buf));
+				janus_strlcat(sofia_log_buf, "\r\n", sizeof(sofia_log_buf));
+			}
+			if(cr != NULL)
+				*cr = '\r';
+			index++;
+		}
+		g_strfreev(parts);
+	} else {
+		return;
+	}
+
+	/* Look for the session this message belongs to */
+	janus_sip_session *session = NULL;
+	janus_mutex_lock(&sessions_mutex);
+	if(strlen(call_id_hdr))
```

Comment:
This should be changed to

    if(strlen(call_id_hdr) > 0)

otherwise I'm not sure if the compiler optimizations discussed in #3059 will kick in.

Reply:
From a cursory check of a few compilers (e.g., https://godbolt.org/z/enodKdhMc) `if(strlen(call_id_hdr))` and `if(strlen(call_id_hdr) > 0)` will compile down to the same check of `call_id_hdr[0] != 0`

So you can use whichever you find more readable.

Reply:
Thanks for checking!